chronos_bin/src/message_processor.rs

```diff
@@ -14,157 +15,157 @@ pub struct MessageProcessor {
-                            headers.insert("readied_by".to_string(), readied_by);
-
-                            tracing::Span::current().record("correlationId", updated_row.id.to_string());
-
-                            publish_futures.push(self.producer.kafka_publish(
-                                updated_row.message_value.to_string(),
-                                Some(headers),
-                                updated_row.message_key.to_string(),
-                                // updated_row.id.to_string(),
-                            ))
-                        }
                         let results = futures::future::join_all(publish_futures).await;
-                        for result in results {
-                            match result {
-                                Ok(m) => {
-                                    tracing::Span::current().record("is_published", "true");
-                                    ids.push(m);
-                                }
-                                Err(e) => {
-                                    tracing::Span::current().record("is_published", "false");
-                                    tracing::Span::current().record("error", &e.to_string());
-
-                                    log::error!("Error: delayed message publish failed {:?}", e);
-                                    break;
-                                    // failure detection needs to pick
-                                }
-                            }
-                        }
+
+                        let ids: Vec<String> = results.into_iter().map(Self::gather_ids).collect();
 
                         if !ids.is_empty() {
-                            if let Err(outcome_error) = &self.data_store.delete_fired_db(&ids).await {
-                                println!("Error: error occurred in message processor delete_fired {}", outcome_error);
-                                //add retry logic here
-                            }
-                            // println!("delete ids {:?} and break", ids);
+                            let _ = self.delete_fired_records_from_db(&ids).await;
+                            log::debug!("number of rows published successfully and deleted from DB {}", ids.len());
                             break;
                         }
-                        log::debug!("number of rows published successfully and deleted from DB {}", ids.len());
-                    } else {
-                        log::debug!("no rows ready to fire for dealine ");
-                        break;
                     }
                 }
                 Err(e) => {
                     if e.contains("could not serialize access due to concurrent update") && retry_count < max_retry_count {
                         //retry goes here
-                        eprintln!("retrying");
+                        log::debug!("retrying");
```

Comment:
Suggestion to always use attempt "N of M" pattern when printing bound iterations.